test/jasmine/tests/heatmap_test.js

```diff
@@ -298,6 +298,11 @@ describe('heatmap calc', function() {
 
         fullTrace._extremes = {};
 
+        // clearCalc used to be (oddly enough) part of supplyDefaults.
```

Comment:
Really? How? `On master` I get:

![image](https://user-images.githubusercontent.com/6675409/49815581-14946480-fd3a-11e8-8a37-8a083fa255b9.png)

Reply:
oh good catch - it wasn't `clearCalc` itself, but [the `setScale` call I removed](https://github.com/plotly/plotly.js/pull/3323/commits/858f7c2aea73f03f06215c280dd11c3429d11584#diff-ad4f76ccd6044ed16514297078e13b84L507) which contains a [fallback for `_categories` and `_categoriesMap`](https://github.com/plotly/plotly.js/blob/3ea5c82de6790a3471fea10774fc50b9c3023ae6/src/plots/cartesian/set_convert.js#L351-L354) - I'll update the comment to describe that. I still don't think we can remove that fallback... though we're closer now.

Reply:
Got it. Thanks for the explanation!

Reply:
Actually looks like we *can* remove that fallback now 🎉  https://github.com/plotly/plotly.js/pull/3323/commits/da30f160b510c88c4c884d901fb37704df52c7e4